py-polars/polars/internals/frame.py

```diff
@@ -4285,49 +4286,97 @@ def first(self) -> DataFrame:
         """
         Aggregate the first values in the group.
         """
-        return self._select_all().first()
```

Comment:
I guess the `_select_all` method can be removed too.

Reply:
I am not entirely sure. We do need to check fi this is still possible.

```python
for sub_df in df.groupby('key'):
   print(sub_df)
```

I hope we tested this. :sweat_smile:

Reply:
We could actually remove `GBSelection` if we no longer supported `GroupBy.__getitem__` as seen here:
```python
from string import ascii_uppercase
import polars as pl

df = pl.DataFrame({'alpha': list(ascii_uppercase) * 8, 'n': range(26 * 8)})
df.groupby('alpha')[['alpha','n']].first()
```

Reply:
Yes. let's remove that as well. I think its better to only support one idomatic syntax.